spot-stub-template.yml

```diff
@@ -0,0 +1,142 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: SPOT stub
+
+Parameters:
+  Environment:
+    Type: String
+    Description: The name of the environment to deploy to
+  CodeSigningConfigArn:
+    Type: String
+    Description: The ARN of the Code Signing Config to use, provided by the deployment pipeline
+    Default: none
+  PermissionsBoundary:
+    Type: String
+    Description: The ARN of the permissions boundary to apply when creating IAM roles
+    Default: none
+
+Mappings:
+  EnvironmentConfiguration:
+    dev:
+      requestQueueArn: "arn:aws:sqs:eu-west-2:816047645251:dev-stub-spot-request-queue"
+      requestQueueUrl: "https://sqs.eu-west-2.amazonaws.com/816047645251/dev-stub-spot-request-queue"
+      requestQueueKeyArn: "arn:aws:kms:eu-west-2:816047645251:key/963c4abe-69e8-4160-a995-1d2b8c054e50" # This will need to change when the key is recreated
+
+Conditions:
+  UseCodeSigning: !Not [!Equals [none, !Ref CodeSigningConfigArn]]
+  UsePermissionsBoundary: !Not [!Equals [none, !Ref PermissionsBoundary]]
+
+Globals:
+  Function:
+    CodeSigningConfigArn: !If
+      - UseCodeSigning
+      - !Ref CodeSigningConfigArn
+      - !Ref AWS::NoValue
+    PermissionsBoundary: !If
+      - UsePermissionsBoundary
+      - !Ref PermissionsBoundary
+      - !Ref AWS::NoValue
+
+Resources:
+  SpotLambda:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub ${Environment}-spot-stub
+      CodeUri: .
+      Handler: src/main/spot-stub/spot.handler
+      Runtime: nodejs20.x
+      Architectures:
+        - arm64
+      Events:
+        Request:
+          Type: SQS
+          Properties:
+            BatchSize: 1
+            Enabled: true
+            Queue: !FindInMap [
+              EnvironmentConfiguration,
+              !Ref Environment,
+              requestQueueArn,
+            ]
```

Comment:
Does this need a `FunctionResponseTypes` of `BatchItemFailures` for batchItemFailures to work correctly?

Reply:
Good spot